File: bindings/node/facade/zenoh.ts
/**
 * Ergonomic facade over the generated Zenoh key-expression binding.
 *
 * A key expression is how a Zenoh network addresses data: a slash-separated
 * path that may carry the `*` and `**` wildcards, so one subscriber names a
 * whole subtree of a fleet rather than each node in it.
 *
 * Only the naming rules cross. Running a Zenoh session needs the std-only zenoh
 * stack, which would land in every install, so it stays in the Rust crate.
 *
 * @packageDocumentation
 */

import {
  keyexprCanonize,
  keyexprIsCanon,
  keyexprIsValid,
  keyexprMatches,
} from '../index'

/** The rules a Zenoh key expression obeys. */
export const keyexpr = {
  /** Reports whether a key expression is well formed. */
  isValid: keyexprIsValid,
  /** Reports whether a key expression is already in its canonical form. */
  isCanon: keyexprIsCanon,
  /**
   * Rewrites a key expression into its canonical form, or `null` if it is
   * malformed.
   *
   * Two expressions that select the same data have one canonical form, so
   * canonizing before comparing or routing avoids treating `a/**\/**\/b` and
   * `a/**\/b` as different.
   */
  canonize: keyexprCanonize,
  /** Reports whether a pattern selects a key. */
  matches: keyexprMatches,
} as const

File: bindings/python/python/pamoja/zenoh.py
"""Idiomatic Zenoh key-expression facade.

A key expression is how a Zenoh network addresses data: a slash-separated path
that may carry the ``*`` and ``**`` wildcards, so one subscriber names a whole
subtree of a fleet rather than each node in it.

Only the naming rules cross. Running a Zenoh session needs the std-only zenoh
stack, which would land in every wheel, so it stays in the Rust crate.
"""

from __future__ import annotations

from ._core import (
    keyexpr_canonize as canonize,
    keyexpr_is_canon as is_canon,
    keyexpr_is_valid as is_valid,
    keyexpr_matches as matches,
)

__all__ = [
    "canonize",
    "is_canon",
    "is_valid",
    "matches",
]
